### PR TITLE
adding a select filter to filter out elements from the json response

### PR DESCRIPTION
**Pull request by vishalchangrani**
Have not yet plugged it in the server.

### CODE REVIEW

File: engine/access/rest/select_filter_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+func ExampleSelectFilter() {
+
+	block, err := generateBlock()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	selectKeys := []string{
+		"header.id",
```
Comment (sideninja) [inline engine/access/rest/select_filter_test.go:20]: Could you add a case for top-level select field such as `payload` and at least one that is non-existing like `foo`?

**Review by sideninja — APPROVED** [review]
Nice :) Maybe it's good if you also integrate it within this branch/PR.

File: engine/access/rest/handler.go [hunk 2]
```diff
@@ -69,6 +80,7 @@ func (h *Handler) jsonResponse(w http.ResponseWriter, response interface{}, logg
 		h.errorResponse(w, http.StatusInternalServerError, "error generating response", logger)
 		return
 	}
+	fmt.Println(string(encodedResponse))
```
Comment (sideninja) [inline engine/access/rest/handler.go:83]: needs to be removed

File: engine/access/rest/select_filter_test.go (hunk 1 above)
Comment (vishalchangrani) [reply engine/access/rest/select_filter_test.go:24]: will add - but note, currently the select logic only supports leaf elements not top-level elements.

Comment (vishalchangrani) [reply engine/access/rest/select_filter_test.go:24]: https://github.com/dapperlabs/flow-go/issues/6034 added this issue and assigned it to myself

**Pull request merged**